### PR TITLE
Set maxBreadcrumbs via Configuration rather than Client

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Provides `config.setMaxBreadcrumbs` in favour of `client.setMaxBreadcrumbs`, which is now deprecated.

## Changeset

- Deprecates `client.setMaxBreadcrumbs` and `Bugsnag.setMaxBreadcrumbs`
- Add `maxBreadcrumbs` property to `Configuration`
- Convert `BreadcrumbsTest` to Kotlin, documenting what each test case attempts to verify, and adding new tests

## Tests

Ran unit tests, mazerunner locally, performed manual testing on dashboard around breadcrumb limits. Added new unit tests for accessor behaviour, and documented existing tests.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [x] Consistency between the changeset and the goal stated above
- [x] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/Breadcrumbs.java [hunk 1]
```diff
@@ -33,32 +38,25 @@ void clear() {
...
+            pruneBreadcrumbs();
         } catch (IOException ex) {
             Logger.warn("Dropping breadcrumb because it could not be serialized", ex);
         }
     }
+
+    private void pruneBreadcrumbs() {
+        int maxBreadcrumbs = configuration.getMaxBreadcrumbs();
+
+        // Remove oldest breadcrumb
```
Comment (simonbowring) [inline sdk/src/main/java/com/bugsnag/android/Breadcrumbs.java]: This comment just needs to be tweaked to state that the oldest breadcrumbs will be removed until the max size is reached, not just one breadcrumb
